vault/audit.go

```diff
@@ -254,16 +256,38 @@ func (c *Core) teardownAudits() error {
 	c.auditLock.Lock()
 	defer c.auditLock.Unlock()
 
+	for _, entry := range c.audit.Entries {
+		c.removeAuditReloadFunc(entry)
+	}
+
 	c.audit = nil
 	c.auditBroker = nil
 	return nil
 }
 
+// removeAuditReloadFunc removes the reload func from the working set. The
+// audit lock needs to be held before calling this.
+func (c *Core) removeAuditReloadFunc(entry *MountEntry) {
+	switch entry.Type {
+	case "file":
+		key := "audit_file|" + entry.Path
+		c.reloadFuncsLock.Lock()
+
+		if c.logger.IsDebug() {
+			c.logger.Debug("audit: removing reload function", "path", entry.Path)
+		}
+
+		delete(c.reloadFuncs, key)
+
+		c.reloadFuncsLock.Unlock()
+	}
```

Comment:
It's purposefully left out in case other types of backends in the future may need reload functionality (or, for instance, there's a reason to change the syslog one).